src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiResponseTypeProvider.cs

```diff
@@ -152,6 +152,8 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
 
                 foreach (var contentType in contentTypes)
```

Comment:
Can you please add a big comment here explaining what the scenarios are and why they work the way that they do. I would have expected this code to look different but there are good reason why it doesn't 😆

Reply:
Done